implementations/rust/ockam/ockam_api/src/nodes/service/tcp_inlets/node_manager.rs

```diff
@@ -109,135 +118,162 @@ impl NodeManager {
+        let inlet = if privileged {
+            #[cfg(privileged_portals_support)]
+            {
+                let (incoming_access_control, outgoing_access_control) =
+                    crate::nodes::service::tcp_inlets::access_control::inlet_access_control(
+                        ctx,
+                        self,
+                        &parameters,
+                        None,
+                    )
+                    .await?;
 
-        let main_replacer: Arc<Mutex<dyn SessionReplacer>> = replacer.clone();
+                // TODO: should options be dependent on the MultiAddr?
+                Arc::new(
+                    self.tcp_transport
+                        .create_privileged_inlet(
+                            listen_addr,
+                            incoming_access_control,
+                            outgoing_access_control,
+                        )
+                        .await?,
+                )
+            }
+            #[cfg(not(privileged_portals_support))]
+            {
+                return Err(ockam_core::Error::new(
+                    Origin::Node,
+                    Kind::Internal,
+                    "Privileged Portals support is not enabled",
+                ));
+            }
+        } else {
+            Arc::new(self.tcp_transport.crate_inlet_multi(listen_addr).await?)
+        };
+
+        let replacer = InletSessionReplacer {
+            node_manager: Arc::downgrade(self),
+            udp_transport,
+            context: ctx.try_clone()?,
+            inlet: inlet.clone(),
+            parameters,
+            status: None,
+        };
 
         let _ = self
             .cli_state
             .create_tcp_inlet(
                 &self.node_name,
                 &listen_addr,
-                &outlet_address,
+                &outlet_addresses,
                 &alias,
                 privileged,
             )
             .await?;
 
-        let additional_session_options = if enable_udp_puncture {
-            Some(AdditionalSessionOptions::create(
-                replacer.clone(),
-                !disable_tcp_fallback,
-            ))
-        } else {
-            None
+        let mut sessions = Vec::with_capacity(target_redundancy + 1);
+        for _ in 0..target_redundancy {
+            // we need replacers with independent status for each session
+            let replacer = Arc::new(Mutex::new(replacer.clone_without_status()?));
+
+            let additional_session_options = if enable_udp_puncture {
+                Some(AdditionalSessionOptions::create(
+                    replacer.clone(),
+                    !disable_tcp_fallback,
+                    ping_timeout,
+                ))
+            } else {
+                None
+            };
+
+            let mut session =
+                Session::create(ctx, replacer, additional_session_options, ping_timeout)?;
+            session.start_monitoring()?;
```

Comment:
Does it make sense to start with a first session instead?

Reply:
We are just spawning new tasks, it should be <<1ms.
If we were to run the first session first, we would delay every other session until the first is established (or fails).